components/vision.py

```diff
@@ -339,7 +340,16 @@ def execute(self) -> None:
 
             if results.multitagResult:
                 self.last_timestamp = timestamp
+
+                bad_tags = [12, 13, 16] if game.is_red() else [1, 2, 3]
+                concatenated_list = bad_tags + results.multitagResult.fiducialIDsUsed
+                if len(set(concatenated_list)) != len(concatenated_list):
```

Comment:
I feel like you just want to check if the intersection of two sets is non-empty? Something like
```suggestion
                bad_tags = BLUE_FIELD_PERIMETER_TAGS if game.is_red() else RED_FIELD_PERIMETER_TAGS
                concatenated_list = bad_tags + results.multitagResult.fiducialIDsUsed
                if set(results.multitagResult.fiducialIDsUsed) & bad_tags:
```

where

```python
BLUE_FIELD_PERIMETER_TAGS = {12, 13, 16}
RED_FIELD_PERIMETER_TAGS = {1, 2, 3}
```